Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,359 @@
+    parser.add_argument('--gcs_locks_path', help='Path to lock repo.')
+    parser.add_argument('--ci_job_id', help='the job id.')
+    parser.add_argument('--test_machines_path', help='the name of the file with all the test machines.')
+    parser.add_argument('--gitlab_status_token', help='gitlab token to get the job status.')
+    parser.add_argument('--response_machine', help='file to update the chosen machine.')
+    parser.add_argument('--lock_machine_name', help='a machine name to lock the specific machine')
+    parser.add_argument('--number_machines_to_lock', help='the needed machines number.', type=int)
+
+    options = parser.parse_args()
+    return options
+
+
+def get_queue_locks_details(storage_client: storage.Client, bucket_name: str, prefix: str):
+    """
+    get a list of all queue locks files.
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        bucket_name(str): the bucket name.
+        prefix(str): the prefix to search for specific files.
+
+    Returns: list of dicts with the job-id and the time_created of the lock file.
+
+    """
+    blobs = storage_client.list_blobs(bucket_name)
+    files = []
+    found = False
+    for blob in blobs:
+        if blob.name.startswith(prefix):
+            found = True
+            files.append({'name': blob.name.strip(prefix), 'time_created': blob.time_created})
+        elif found:
+            break
+    return files
+
+
+def get_machines_locks_details(storage_client: storage.Client, bucket_name: str,
+                               lock_repository_name: str, machines_lock_repo: str):
+    """
+    get a list of all machines locks files.
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        bucket_name(str): the bucket name.
+        lock_repository_name(str):  the lock_repository_name name.
+        machines_lock_repo(str): the machines_lock_repo name.
+
+    Returns: list of dicts with the job-id and the time_created of the lock file.
+
+    """
+    blobs = storage_client.list_blobs(bucket_name)
+    files = []
+    found = False
+    for blob in blobs:
+        if blob.name.startswith(lock_repository_name):
+            found = True
+            if blob.name.startswith(f'{lock_repository_name}/{machines_lock_repo}'):
+                lock_file_name = blob.name.split('/')[-1]
+                if lock_file_name:
+                    files.append({'machine_name': lock_file_name.split('-lock-')[0],
+                                  'job_id': lock_file_name.split('-lock-')[1]})
+        elif found:
+            break
+    return files
+
+
+def check_job_status(token: str, job_id: str):
+    """
+    get the status of a job in gitlab.
+    Args:
+        token(str): the gitlab token.
+        job_id(str): the job id to check.
+    Returns: the status of the job.
+
+    """
+    user_endpoint = f"https://code.pan.run/api/v4/projects/2596/jobs/{job_id}"  # disable-secrets-detection
```

Comment:
Please add as constant. 
JOB_STATUS_URL = https://code.pan.run/api/v4/projects/{}/jobs/{}
CONTENT_GITLAB_PROJECT_ID = 2596
```suggestion
    user_endpoint = JOB_STATUS_URL.format(CONTENT_GITLAB_PROJECT_ID, job_id)  # disable-secrets-detection
```